cmd/do-agent/config.go

```diff
@@ -154,7 +181,15 @@ func initCollectors() []prometheus.Collector {
 	}
 
 	if config.kubernetes != "" {
-		k, err := collector.NewScraper("dokubernetes", config.kubernetes, k8sWhitelist, defaultTimeout)
+		kubernetesClusterUUID, err := getKubernetesClusterUUID()
+		if err != nil {
+			log.Error("Failed to get cluster UUID when initializing DO Kubernetes metrics: %+v", err)
+		}
+		var kubernetesLabels []*dto.LabelPair
+		if kubernetesClusterUUID != "" {
```

Comment:
should we maybe just exit here if this is empty?

Reply:
I was thinking the same thing. I think that if the UUID is empty we should abort the process altogether since it's claiming to be a kubernetes node and it doesn't appear to be true.

Reply:
but after talking with Nan yesterday with Thor, we can't really do that because legacy clusters will not have the UUID. If we want to still gather k8s metrics for those clusters we need to allow them to still be tied to host_id in the interim